Scripts/script-ParsePCAP.yml

```diff
@@ -0,0 +1,294 @@
+        getvals = operator.itemgetter('source_ip','destination_ip', 'uri')
+        self._http_request_data.sort(key=getvals)
+
+        result = []
+        for k, g in itertools.groupby(self._http_request_data, getvals):
+            result.append(g.next())
+
+        self._http_request_data[:] = result
+        return self._http_request_data
+
+    def get_dns_request_data(self):
+        getvals = operator.itemgetter('type','request', 'response')
+        self._dns_request_data.sort(key=getvals)
+
+        result = []
+        for k, g in itertools.groupby(self._dns_request_data, getvals):
+            result.append(g.next())
+
+        self._dns_request_data[:] = result
+        return self._dns_request_data
+
+    def get_flows(self):
+        return self.quick_unique(self._flows)
+
+    def get_packet_count(self):
+        return self._packet_count
+
+    def get_http_count(self):
+        return self._http_count
+
+    def get_dns_count(self):
+        return self._dns_count
+
+    def summary2json(self):
+        self._dns_json = []
+        self._dest_json = []
+        self._http_json = []
+        self._flow_json = []
+        self._counts_json = []
+
+        self._counts_json.append({"packet_count":self._packet_count,"http_count":self._http_count,"dns_count":self._dns_count})
+
+        for dns in self.get_dns_request_data():
+            self._dns_json.append({"type": dns["type"], "request": dns["request"], "response": dns["response"]})
+
+        for ip in self.get_destination_ip_details():
+            self._dest_json.append({"ip_address" : ip['ip_address'], "owner": ip['owner'], "asn": ip['asn'], "netblock": ip['block']})
+
+        for info in self.get_http_request_data():
+            tmp = {}
+            for key, value in info.items():
+                tmp[key] = value
+            self._http_json.append(tmp)
+
+        for flow in self.get_flows():
+            direction = flow.split("=>")
+            source = direction[0].split("/")
+            destination = direction[1].split("/")
+            self._flow_json.append({"source_ip":source[0],"source_port":source[1],"destination_ip":destination[0],"destination_port":destination[1]})
+
+        #construct the output object
+        obj = {"file":self._pcap_file,
+               "dns_data":self._dns_json,
+               "destination_ip_details":self._dest_json,
+               "http_requests":self._http_json,
+               "flows":self._flow_json,
+               "counts":self._counts_json
+               }
+
+        return json.dumps(obj)
+
+
+
+filePath = demisto.executeCommand('getFilePath', {'id': demisto.args()['entryId'] })
+readyPath = filePath[0]["Contents"]["path"]
+#demisto.log(filePath)
+#file = open(readyPath, 'rb')
+#data=file.read()
+#demisto.log(readyPath)
+#demisto.results( { "Type" : entryTypes["note"], "ContentsFormat" : formats["text"], "Contents" : "Testing" } )
+ret = []
+filename = filePath[0]["Contents"]["name"]
+demisto.log('PCAP FILE PROCESSED')
```

Comment:
also logs should be deleted if you finished, because it will be printed to war room